src/states/elder/tests.rs

```diff
@@ -544,7 +544,11 @@ struct JoiningPeer {
 
 impl JoiningPeer {
     fn new(rng: &mut MainRng) -> Self {
-        let (network_event_tx, network_event_rx) = crossbeam_channel::unbounded();
+        let (network_event_tx, network_event_rx) = {
+            let (node_tx, node_rx) = crossbeam_channel::unbounded();
+            let (client_tx, _) = crossbeam_channel::unbounded();
+            (quic_p2p::EventSenders { node_tx, client_tx }, node_rx)
+        };
```

Comment:
I explicitly did not make EventReceivers part of the API.
We have to do it because for tests we create it all at once, but in production we do not.
I considered created a helper function in tests, but it did not seem to bring much here.

Reply:
> but in production we do not.

Not sure what you mean by that, because in the above code you seem to be creating both channels at once anyway.